src/NzbDrone.Core/Messaging/Commands/CommandExecutor.cs

```diff
@@ -1,117 +1,65 @@
 using System;
-using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using NLog;
 using NzbDrone.Common;
-using NzbDrone.Common.EnsureThat;
-using NzbDrone.Common.Serializer;
 using NzbDrone.Common.TPL;
-using NzbDrone.Core.Messaging.Commands.Tracking;
+using NzbDrone.Core.Lifecycle;
 using NzbDrone.Core.Messaging.Events;
 using NzbDrone.Core.ProgressMessaging;
 
 namespace NzbDrone.Core.Messaging.Commands
 {
-    public class CommandExecutor : ICommandExecutor
+    public class CommandExecutor : //IHandle<CommandQueuedEvent>,
+                                   IHandle<ApplicationStartedEvent>,
+                                   IHandle<ApplicationShutdownRequested>
     {
         private readonly Logger _logger;
         private readonly IServiceFactory _serviceFactory;
-        private readonly ITrackCommands _trackCommands;
+        private readonly ICommandService _commandService;
         private readonly IEventAggregator _eventAggregator;
         private readonly TaskFactory _taskFactory;
 
-        public CommandExecutor(Logger logger, IServiceFactory serviceFactory, ITrackCommands trackCommands, IEventAggregator eventAggregator)
+        private static CancellationTokenSource _cancellationTokenSource;
+        private const int THREAD_LIMIT = 1;
```

Comment:
> Actually, CheckForFinishedDownload & Rss Sync is a command on it's own. Wouldn't that be run on 1 thread as well?

Not following exactly what you mean. Each command will be single threaded, but we need to avoid running multiple add series commands at the same time (maybe its a single job and has its own queue?)